### PR TITLE
[TwigBundle] Reconfigure twig paths when they are updated

### PR DESCRIPTION
**Pull request by chbruyand**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #14771, #14768, #14262, #14778 |
| License | MIT |

Refresh twig paths upon creation and deletion. As we don't care neither about path's modification time nor path's content, a new Resource has been added in the Config Component.
Full discussion in #14778.

### CODE REVIEW

**Comment by mpdude** [comment]
Not sure, but does `src/Symfony/Bundle/TwigBundle/composer.json` needs an update for `symfony/config`?

**Comment by chbruyand** [comment]
I think so, but don't know which version to put there

**Comment by mpdude** [comment]
I'd go for ~2.8, which is the first version that will contain the new resource.

File: src/Symfony/Component/Config/Resource/FileExistenceResource.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        $this->resource = $resource;
+        $this->exists = file_exists($resource);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __toString()
+    {
+        return (string) $this->resource;
```
Comment (sstok) [inline src/Symfony/Component/Config/Resource/FileExistenceResource.php]: You can do the string casting in the constructor already.

**Comment by aitboudad** [comment]
:+1:

**Comment by chbruyand** [comment]
Any other suggestion / improvement ?

File: src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php [hunk 2]
```diff
@@ -74,23 +75,30 @@ public function load(array $configs, ContainerBuilder $container)
             } else {
                 $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($path, $namespace));
             }
+            $container->addResource(new FileExistenceResource($path));
```
Comment (stof) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: not needed here as we don't change the container configuration based on the existence of this folder

Comment (chbruyand) [inline src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php]: Thanks for the hint

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @chbruyand.

**Pull request closed without merging**